### ISSUE TITLE
Decouple GraphQL and Plugins

### ISSUE THREAD

**Issue by illegalprime** [HackGT/buzzer#8]
This is to make adding a plugin and maintaining it simpler.
Each plugin should have its own snippet of GraphQL that defines its configuration options, e.g. slack:
```GraphQL
type Config {
    channels: [String!]!
    at_channel: Boolean!
    at_here: Boolean!
}
```

The server should collect all of these config options and stich them together to make a final GraphQL schema. GraphQL provides utilities to do this, but you might have to get a bit hacky to rename `Config` to a unique name for that plugin and to collect them all in one struct.

The return status for each plugin should probably stay the same.

**Comment by illegalprime** [comment]
This could help: https://www.npmjs.com/package/merge-graphql-schemas

**Comment by illegalprime** [comment]
Fixed by #21
